### PR TITLE
fix: make `useBuildkit` field nullable across all config versions

### LINKED ISSUE #4936: Enable Docker BuildKit by default
With Docker Desktop 2.4.0.0, BuildKit is now enabled by default (release notes for [_Docker Desktop for Windows_](https://docs.docker.com/docker-for-windows/release-notes/) and [_Docker Desktop for Mac_](https://docs.docker.com/docker-for-mac/release-notes/)):
> Docker Desktop now enables BuildKit by default after a reset to factory defaults.

If Docker is confident enough in its use, then we should enable it by default too!

### PR DESCRIPTION
**Pull request by gsquared94**
fix: setting `useBuildkit: false` actually turns off buildkit for docker builder

**Related**: #4936

**Description**

Currently setting `useBuildkit: false` in the `skaffold.yaml` config doesn't actually disable buildkit. This PR fixes that by setting `DOCKER_BUILDKIT =0` env variable when `useBuildkit: false` is set in the config.

In `pkg/skaffold/build/docker/docker.go `:
```diff
-	if b.useBuildKit {
-		cmd.Env = append(cmd.Env, "DOCKER_BUILDKIT=1")
+	if b.useBuildKit != nil {
+		if *b.useBuildKit {
+			cmd.Env = append(cmd.Env, "DOCKER_BUILDKIT=1")
+		} else {
+			cmd.Env = append(cmd.Env, "DOCKER_BUILDKIT=0")
+		}
	}
``` 

This change in itself would however break backwards compatibility since for previous versions of the config even if `useBuildkit` property is not set, it'd default to `false` and now start disabling buildkit which is not how it worked previously (so that bug essentially became a feature, that if `useBuildkit` is unspecified it should use the Docker default).

So we additionally make the config change to all config versions, so that a missing `useBuildkit` field can be identified separately from an explicitly set `useBuildkit: false`

**User facing changes (remove if N/A)**
There are no user facing changes other than if users were setting `useBuildkit: false` in the config, skaffold will now actually disable buildkit.

**Follow-up Work (remove if N/A)**

Fix #4936 after this is merged.

> Changes discussed with @briandealwis 

> The `make checks` will be failing since we generally prohibit making changes to already released configs. But this PR only changes a single field from `bool` to `*bool`.

### CODE REVIEW

File: pkg/skaffold/build/docker/docker.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, a *latestV1.Artifact
 
 	var imageID string
 
-	if b.useCLI || b.useBuildKit {
+	if b.useCLI || (b.useBuildKit != nil && *b.useBuildKit) {
```
Comment (briandealwis) [inline pkg/skaffold/build/docker/docker.go:57]: Worth a comment here to say that we ignore the useCLI setting if buildkit is requested as we currently use the Docker CLI to use buildkit. We could consider a different approach in the future.

File: pkg/skaffold/build/docker/docker_test.go [hunk 2]
```diff
@@ -135,7 +135,7 @@ func TestDockerCLIBuild(t *testing.T) {
 					test.err,
 				)
 				t.Override(&util.DefaultExecCommand, mockCmd)
-			} else if test.localBuild.UseBuildkit || test.localBuild.UseDockerCLI {
+			} else if (test.localBuild.UseBuildkit != nil && *test.localBuild.UseBuildkit) || test.localBuild.UseDockerCLI {
```
Comment (briandealwis) [inline pkg/skaffold/build/docker/docker_test.go:138]: This feels like it should be a `want` field in the test: `wantDockerCLI`.

Comment (briandealwis) [inline pkg/skaffold/build/docker/docker_test.go:138]: This feels like it should be a `want` field in the test: `wantDockerCLI`.

**Review by briandealwis — APPROVED** [review]

File: pkg/skaffold/build/docker/docker_test.go (hunk 2 above)
Comment (gsquared94) [reply pkg/skaffold/build/docker/docker_test.go]: done.

Comment (gsquared94) [reply pkg/skaffold/build/docker/docker_test.go]: done

**Review by nkubala — APPROVED** [review]

**Pull request merged**